src/kit_version.rs

```diff
@@ -0,0 +1,207 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::ser::{Serialize, Serializer};
+use std::convert::TryFrom;
+use std::fmt;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct BuildInfo {
+    pub commit: String,
+    pub branch: Option<String>,
+}
+
+impl fmt::Display for BuildInfo {
```

Comment:
Suggestion: Rather than implementing `Display`, it might be nicer to implement `ToString`, and have that be called from `Display`

Reply:
From https://doc.rust-lang.org/std/string/trait.ToString.html:

> This trait is automatically implemented for any type which implements the Display trait. As such, ToString shouldn’t be implemented directly: Display should be implemented instead, and you get the ToString implementation for free.

If you try to implement `Display` and `ToString`, there are conflicting implementations.